server/channels/app/migrations.go

```diff
@@ -556,6 +556,21 @@ func (s *Server) doPostPriorityConfigDefaultTrueMigration() {
 }
 
 func (s *Server) doElasticsearchFixChannelIndex(c *request.Context) {
+	s.AddLicenseListener(func(oldLicense, newLicense *model.License) {
+		if model.BuildEnterpriseReady != "true" || newLicense == nil || !*newLicense.Features.Elasticsearch {
+			return
+		}
+
+		if _, err := s.Store().System().GetByName(model.MigrationKeyElasticsearchFixChannelIndex); err == nil {
+			return
+		}
+
+		if _, appErr := s.Jobs.CreateJob(c, model.JobTypeElasticsearchFixChannelIndex, nil); appErr != nil {
+			mlog.Fatal("failed to start job for fixing Elasticsearch channels index on license application", mlog.Err(appErr))
+			return
+		}
```

Comment:
This looks like the same code is being repeated again. Perhaps we can refactor to a separate function?